src/spikeinterface/sortingcomponents/peak_selection.py

```diff
@@ -66,8 +71,24 @@ def select_peaks(peaks, method="uniform", seed=None, return_indices=False, **met
         return_indices is True.
     """
 
+    if margin is not None:
+        assert recording is not None, "recording should be provided if margin is not None"
+
     selected_indices = select_peak_indices(peaks, method=method, seed=seed, **method_kwargs)
     selected_peaks = peaks[selected_indices]
+
+    if margin is not None:
```

Comment:
I have the intuition that we could make a simple while loop directly on selectied_indices to avoid this full spikes vector masking twice and then boolean indexing again.
No ?